### PR TITLE
MEN-4996 - filter by monitoring issues

### PR DESCRIPTION
**Pull request by mzedel**
Changelog: None
Signed-off-by: Manuel Zedel <manuel.zedel@northern.tech>

### CODE REVIEW

File: src/js/constants/deviceConstants.js [hunk 1]
```diff
@@ -109,6 +109,35 @@ module.exports = {
...
+          yesterday.setDate(yesterday.getDate() - 1);
+          return yesterday.toISOString();
+        }
+      },
+      title: 'Offline devices'
+    },
+    monitoring: {
+      key: 'monitoring',
+      needsReporting: false,
+      filterRule: { scope: 'monitoring', key: 'alerts', operator: '$gt', value: 0 },
```
Comment (tranchitella) [inline src/js/constants/deviceConstants.js:131]:
```suggestion
      filterRule: { scope: 'monitoring', key: 'alerts', operator: '$eq', value: true },
```

File: src/js/constants/deviceConstants.js [hunk 2]
```diff
@@ -109,6 +109,35 @@ module.exports = {
...
+      title: 'Devices reporting monitoring issues'
+    },
+    authRequests: {
+      key: 'authRequests',
+      needsReporting: true,
+      filterRule: { scope: 'monitoring', key: 'auth_requests', operator: '$gt', value: 1 },
+      title: 'Devices with new authentication requests'
+    }
+  },
```
Comment (tranchitella) [inline src/js/constants/deviceConstants.js:140]: This is not implemented (as of today). In any case, it wouldn't be `monitoring` scope.

Comment (mzedel) [reply src/js/constants/deviceConstants.js:140]: In order to provide the functionality as it is defined in the design we should get it implemented though... But once we get the reporting service in we can likely adjust a few places anyway (e.g. finally getting #992 in)...

Comment (tranchitella) [reply src/js/constants/deviceConstants.js:140]: @mzedel does this block the release? we don't have ANY plan today to implement this feature in the backend before the introduction of the reporting service.

Comment (mzedel) [reply src/js/constants/deviceConstants.js:140]: nono, that doesn't block at all - this (and the offline one) are just in preparation (to check the general approach) and are filtered based on the `needsReporting: true` 👌

**Review by tranchitella — APPROVED** [review]
Looks okay, and works fine. The only (minor) thing which annoys me is that it makes the request to count the number of devices with +1 auth sets despite the fact reporting is not enabled. Can you double-check and make sure the "count" API request is not fired in that case?

**Comment by mzedel** [comment]
> Looks okay, and works fine. The only (minor) thing which annoys me is that it makes the request to count the number of devices with +1 auth sets despite the fact reporting is not enabled. Can you double-check and make sure the "count" API request is not fired in that case?

Thanks for going through this @tranchitella! Sorry I missed that - ab71ebba7039d8333efff01a5c71d243b6fb9f2d includes the adjustment and shouldn't call before there is reporting support 👍

**Review by tranchitella — APPROVED** [review]

**Comment by mzedel** [comment]
@tranchitella the device list should now show the monitoring notifications 👌 - please take a look at the added commits if you have the time and I'll take care of the update on staging once you give the green light

**Comment by tranchitella** [comment]
@mzedel it works fine, the only minor issue is that on my screen the monitoring badge is not fully visible, see screenshot:

image

**Comment by mzedel** [comment]
Yeah, there is MEN-4988 and I'll take care of it in that context asap... This should be pre-existing though and appear with pending devices too - I'll merge regardless and try to improve it before the end of the sprint.

**Pull request merged**